src/structures/ShardManager.ts

```diff
@@ -0,0 +1,19 @@
+import DiscordWebsocket from '../../ws/DiscordWebsocket';
+import Collection from '../util/Collection';
+
+export default class ShardManager {
+  shards: Collection<DiscordWebsocket>;
+
+  private shardID: number;
+
+  constructor() {
+    this.shards = new Collection({ base: DiscordWebsocket });
+    this.shardID = 0;
+  }
+
+  connect(websocket: DiscordWebsocket) {
```

Comment:
Name this register and take a shard ID as a parameter, instantiate DiscordWebsocket inside the method